### PR TITLE
Optimize "ballerina run" command execution flow

### LINKED ISSUE #20841: Optimize "ballerina run" command execution flow
**Description:**
Current implementation for `ballerina run` command can be further optimized by removing redundant phases(i.e. final uber jar creation) so that the program startup time will be improved.

**Steps to reproduce:**

**Affected Versions:**
jBallerina 1.1.1 and below.

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
Fixes #20841.

## Approach
This PR excludes `CreateExecutableTask` phase (where the final uber jar is created) from the `ballerina run` execution flow. Therefore instead of creating a single executable uber jar and executing it, the executable thin jar will be running with all its dependency jars added to the classpath. Therefore this improves the ballerina program startup time. 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java [hunk 1]
```diff
@@ -207,15 +195,16 @@ private void runGeneratedExecutableWithSameClassLoader(BLangPackage executableMo
...
-        }
-        StringBuilder classPath = new StringBuilder(jar.moduleJar.toString());
-        for (Path path : jar.platformLibs) {
-            classPath.append(seperator).append(path);
-        }
-        return classPath.toString();
+    private String getAllClassPaths(BLangPackage executableModule, BuildContext buildContext) {
+        // Since classpath separator depends on the OS type
+        String separator = OsUtils.isWindows() ? ";" : ":";
+        StringBuilder cp = new StringBuilder();
```
Comment (rasika) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: You can use `StringJoiner` with `File.pathSeparator` here.

`StringJoiner joiner = new StringJoiner(File.pathSeparator);`

Comment (NipunaRanasinghe) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: @rasika That's a good suggestion! I did a rough timing analysis for both approaches and `StringJoiner` is slighly faster than StringBuilder in here. Thanks for pointing that out :)

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java [hunk 2]
```diff
@@ -64,71 +65,59 @@
...
-                        throw createLauncherException("no entry points found in '" + singleFileContext.getBalFile() +
-                                "'.");
-                    case SINGLE_MODULE:
-                        SingleModuleContext singleModuleContext = buildContext.get(BuildContextField.SOURCE_CONTEXT);
-                        throw createLauncherException("no entry points found in '" +
-                                singleModuleContext.getModuleName() + "'.");
-                    default:
-                        throw createLauncherException("unknown source type found when running executable.");
-                }
+        // check if any entry points are found.
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: not found?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java [hunk 3]
```diff
@@ -64,71 +65,59 @@
...
-
-        // if the executable is not a file and not an extension with .jar
-        if (!(Files.isRegularFile(this.executablePath) &&
-                this.executablePath.toString().endsWith(BLANG_COMPILED_JAR_EXT))) {
-
-            throw createLauncherException("cannot run '" + this.executablePath.toAbsolutePath().toString() +
-                    "' as it is not an executable with .jar extension.");
+        // if the executable is not a file and not an extension with .jar.
+        if (!(Files.isRegularFile(this.executableJarPath) &&
+                this.executableJarPath.toString().endsWith(BLANG_COMPILED_JAR_EXT))) {
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: Do we need this check. Executable always jar file

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java [hunk 4]
```diff
@@ -53,7 +54,7 @@
 public class RunExecutableTask implements Task {
 
     private final String[] args;
-    private Path executablePath;
+    private Path executableJarPath;
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: this is module jar i think?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java (hunk 3 above)
Comment (NipunaRanasinghe) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: I added the same check we had for the final uber jar earlier. If we dont need this for the module jar, we can remove.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java (hunk 4 above)
Comment (NipunaRanasinghe) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: Yeah this is the module executable jar file path. Do you have any alternative suggestions for the var name?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java (hunk 3 above)
Comment (NipunaRanasinghe) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: Removing the check as per the offline discussion.

**Pull request merged**